### ISSUE TITLE
Issues Running on MacOS

### ISSUE THREAD

**Issue by frmsaul** [HearthSim/python-bnet#1]
Hi, 

Im trying to run this on MacOS but when I run the `bootstrap.sh` script I get this error:

```
Fetching data files from https://github.com/HearthSim/hs-proto.git into /Users/saulfuhrmann/Computers/EdenBnet/python-bnet/hs-proto
HEAD is now at 81ae2ab Update to patch 4.2.0.12051
pegasus/shared.proto: File recursively imports itself: pegasus/shared.proto -> pegasus/util.proto -> pegasus/shared.proto
pegasus/util.proto: Import "pegasus/shared.proto" was not found or had errors.
pegasus/util.proto:190:20: Extension range 100 to 10000 includes field "scenario_asset" (100).
pegasus/util.proto:73:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:86:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:87:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:98:18: "PegasusShared.ProfileNoticeMedal" is not defined.
pegasus/util.proto:99:18: "PegasusShared.ProfileNoticeRewardBooster" is not defined.
pegasus/util.proto:100:18: "PegasusShared.ProfileNoticeRewardCard" is not defined.
pegasus/util.proto:101:18: "PegasusShared.ProfileNoticePreconDeck" is not defined.
pegasus/util.proto:102:18: "PegasusShared.ProfileNoticeRewardDust" is not defined.
pegasus/util.proto:103:18: "PegasusShared.ProfileNoticeRewardGold" is not defined.
pegasus/util.proto:104:18: "PegasusShared.ProfileNoticeRewardMount" is not defined.
pegasus/util.proto:105:18: "PegasusShared.ProfileNoticeRewardForge" is not defined.
pegasus/util.proto:108:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:109:18: "PegasusShared.ProfileNoticePurchase" is not defined.
pegasus/util.proto:110:18: "PegasusShared.ProfileNoticeCardBack" is not defined.
pegasus/util.proto:111:18: "PegasusShared.ProfileNoticeDisconnectedGameResult" is not defined.
pegasus/util.proto:112:18: "PegasusShared.ProfileNoticeBonusStars" is not defined.
pegasus/util.proto:113:18: "PegasusShared.ProfileNoticeAdventureProgress" is not defined.
pegasus/util.proto:114:18: "PegasusShared.ProfileNoticeLevelUp" is not defined.
pegasus/util.proto:115:18: "PegasusShared.ProfileNoticeAccountLicense" is not defined.
pegasus/util.proto:119:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:120:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:125:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:161:18: "PegasusShared.GameType" is not defined.
pegasus/util.proto:170:18: "PegasusShared.ProfileNoticeRewardBooster" is not defined.
pegasus/util.proto:171:18: "PegasusShared.ProfileNoticeRewardCard" is not defined.
pegasus/util.proto:172:18: "PegasusShared.ProfileNoticeRewardDust" is not defined.
pegasus/util.proto:173:18: "PegasusShared.ProfileNoticeRewardGold" is not defined.
pegasus/util.proto:174:18: "PegasusShared.ProfileNoticeRewardMount" is not defined.
pegasus/util.proto:175:18: "PegasusShared.ProfileNoticeRewardForge" is not defined.
pegasus/util.proto:187:18: "PegasusShared.AssetKey" is not defined.
pegasus/util.proto:188:18: "PegasusShared.DatabaseResult" is not defined.
pegasus/util.proto:189:18: "PegasusShared.ScenarioDbRecord" is not defined.
pegasus/util.proto:200:18: "PegasusShared.DeckCardData" is not defined.
pegasus/util.proto:201:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:309:18: "PegasusShared.DeckType" is not defined.
pegasus/util.proto:398:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:481:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:498:18: "PegasusShared.Platform" is not defined.
pegasus/util.proto:667:18: "PegasusShared.BattlePayProvider" is not defined.
pegasus/util.proto:680:18: "PegasusShared.FavoriteHero" is not defined.
pegasus/util.proto:689:18: "PegasusShared.AssetKey" is not defined.
pegasus/util.proto:764:18: "PegasusShared.PlayQueueInfo" is not defined.
pegasus/util.proto:837:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:885:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:894:18: "PegasusShared.RewardChest" is not defined.
pegasus/util.proto:912:18: "PegasusShared.DeckCardData" is not defined.
pegasus/util.proto:915:18: "PegasusShared.RewardChest" is not defined.
pegasus/util.proto:916:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:917:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:926:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:927:18: "PegasusShared.CardDef" is not defined.
pegasus/util.proto:954:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:972:18: "PegasusShared.DeckInfo" is not defined.
pegasus/util.proto:980:18: "PegasusShared.CardStack" is not defined.
pegasus/util.proto:989:18: "PegasusShared.DeckCardData" is not defined.
pegasus/util.proto:997:18: "PegasusShared.DatabaseAction" is not defined.
pegasus/util.proto:998:18: "PegasusShared.DatabaseResult" is not defined.
pegasus/util.proto:1007:18: "PegasusShared.DeckInfo" is not defined.
pegasus/util.proto:1032:18: "PegasusShared.BoosterInfo" is not defined.
pegasus/util.proto:1139:18: "PegasusShared.BattlePayProvider" is not defined.
pegasus/util.proto:1163:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:1171:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:1214:18: "PegasusShared.Date" is not defined.
pegasus/util.proto:1285:18: "PegasusShared.AdventureProgress" is not defined.
pegasus/util.proto:1327:18: "PegasusShared.TavernBrawlSpec" is not defined.
pegasus/util.proto:1336:18: "PegasusShared.TavernBrawlPlayerRecord" is not defined.
pegasus/util.proto:1344:18: "PegasusShared.FavoriteHero" is not defined.
pegasus/util.proto:1353:18: "PegasusShared.FavoriteHero" is not defined.
pegasus/util.proto:1379:18: "PegasusShared.AccountLicenseInfo" is not defined.
pegasus/shared.proto: Import "pegasus/util.proto" was not found or had errors.
pegasus/shared.proto:403:20: Extension range 100 to 10000 includes field "my_record" (105).
pegasus/shared.proto:365:16: ".PegasusUtil.AssetResponse" is not defined.
pegasus/shared.proto:373:16: ".PegasusUtil.AssetResponse" is not defined.
pegasus/game.proto: Import "pegasus/shared.proto" was not found or had errors.
pegasus/game.proto:34:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:49:18: "PegasusShared.Platform" is not defined.
pegasus/game.proto:114:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:115:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:123:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:301:18: "PegasusShared.Platform" is not defined.
pegasus/game.proto:302:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:323:18: "PegasusShared.BnetId" is not defined.
pegasus/game.proto:334:18: "PegasusShared.BnetId" is not defined.
```

Any idea what the issue might be?

**Comment by Bert-Proesmans** [comment]
Hi

tl;dr: The project is outdated, any help is appreciated! :)

The issue here is outdated protobuffer files since the `hs-proto` repository hasn't been updated in a long time. It's not an OS specific issue.
Protobuffer files are extracted by our [Proto-extractor tool](https://github.com/HearthSim/proto-extractor), which is updated to work around the circular import problems. But the folder structure of the extracted proto files changed as a consequence. This has affected all related projects and some, like the one in this repo, have not been made compatible with the new protobuffer files.

Solution:
You can extract and compile proto files yourself using the proto-extractor (if you have hearthstone installed). With the latest compiled protobuffer files you'll need to adjust some imports (and possibly a bit of logic) to get this project to compile.

@jleclanche developed the code of this project; he's able to provide implementation details, but he is also busy with other projects.

Bert

**Comment by EionRobb** [comment]
@Bert-Proesmans Can you point in the direction of projects that *have* had their proto's updated?

**Comment by jleclanche** [comment]
@EionRobb Are you still around on Discord? We can chat about the proto extraction. I'd like to get the hs-proto repository going again actually.
